ext/afform/core/afform.php

```diff
@@ -127,6 +127,33 @@ function afform_civicrm_upgrade($op, CRM_Queue_Queue $queue = NULL) {
  */
 function afform_civicrm_managed(&$entities) {
   _afform_civix_civicrm_managed($entities);
+
+  $afforms = \Civi\Api4\Afform::get(FALSE)
+    ->setSelect(['title', 'name', 'module_name', 'directive_name'])
+    ->addWhere('is_dashlet', '=', TRUE)
+    ->execute();
+  $afforms = $afforms->getArrayCopy();
```

Comment:
If calling the API from within this hook is a problem, you could just use the Afform BAO.

Reply:
Good point. I'll give it a try. (And even if it doesn't work, it might give a more enlightening backtrace.)

Reply:
This was a good tip. Skipping down a layer (`s/APIv4/BAO/`; `s/BAO/afform_scanner/`) helped to find the quirky condition that arose during test init. That had a similar problem, but it was easier to debug/workaround.